lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
         return this.kuzzle.services.list.internalCache.pexpire(cacheKey, ttl);
       })
       .then(() => this.client.scroll(esRequest))
-      .then(result => formatSearchResults(result))
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .then(({ body }) => {
+        const apiResponse = {
+          scrollId: body.scrollId,
+          hits: body.hits,
+          total: body.total
+        };
+
+        return apiResponse;
+      })
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Search documents from elasticsearch with a query
-   * @param {Request} request
-   * @returns {Promise} resolve documents matching the filter
+   *
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {object} body - Request body (query, sort, etc.)
+   * @param {object} options - from (null), size (null), scroll (null)
+   *
+   * @returns {Promise} { scrollId, hits, aggregations, total }
    */
-  search(request) {
-    const esRequest = initESRequest(request, ['from', 'size', 'scroll']);
-
-    esRequest.body = request.input.body;
-
-    if (!request.input.args.includeTrash) {
-      addActiveFilter(esRequest);
-    }
+  search (index, collection, body, { from=null, size=null, scroll=null }) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body,
+      from,
+      size,
+      scroll
+    };
 
     return this.client.search(esRequest)
-      .then(result => {
-        const formatted = formatSearchResults(result);
+      .then(({ body }) => {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') "body" hides or potentially hides a variable declared in an outer scope at line 191. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AVariableShadowing)